vm/types.go

```diff
@@ -6,41 +6,36 @@ import (
 	iotago "github.com/iotaledger/iota.go/v4"
 )
 
-type OutputWithCreationSlot struct {
-	Output       iotago.Output
-	CreationSlot iotago.SlotIndex
-}
-
-// InputSet is a map of OutputID to OutputWithCreationSlot.
-type InputSet map[iotago.OutputID]OutputWithCreationSlot
+// InputSet is a map of OutputID to Output.
+type InputSet map[iotago.OutputID]iotago.Output
 
 func (inputSet InputSet) OutputSet() iotago.OutputSet {
 	outputs := make(iotago.OutputSet, len(inputSet))
 	for outputID := range inputSet {
-		outputs[outputID] = inputSet[outputID].Output
+		outputs[outputID] = inputSet[outputID]
 	}
 
 	return outputs
 }
 
-type ChainOutputWithCreationSlot struct {
-	ChainID      iotago.ChainID
-	Output       iotago.ChainOutput
-	CreationSlot iotago.SlotIndex
+type ChainOutput struct {
```

Comment:
I think this should be renamed to ChainOutputWithOutputID. It is a little confusing when we use this type (vm.ChainOutput) for the inputs and iotago.ChainOutput on the output side in some places in the VM.